File: backend/src/DineOS.Domain/Entities/Payment.cs
using DineOS.Domain.Common;
using DineOS.Domain.Enums;

namespace DineOS.Domain.Entities;

public class Payment : TenantAuditingEntity
{
    public long OrderId { get; set; }
    public decimal Amount { get; set; }
    public PaymentMethod Method { get; set; }
    public PaymentStatus Status { get; set; } = PaymentStatus.Completed;

    /// <summary>
    /// Set when the OverduePaymentNotificationJob has emailed the owner about
    /// this still-Pending payment. Stops repeat notifications.
    /// </summary>
    public DateTime? OverdueNotifiedAt { get; set; }
}

File: frontend/src/types/payment.ts
export type PaymentMethod = "Cash" | "Card";

export type PaymentStatus = "Completed" | "Refunded" | "Pending";

export interface Payment {
  id: string;
  orderId: string;
  amount: number;
  method: PaymentMethod;
  status: PaymentStatus;
  tenantId?: string;
  createdAt: string;
}

export interface ProcessPaymentInput {
  orderId: string;
  amount: number;
  method: PaymentMethod;
}
